common/test_filestore_utils.py

```diff
@@ -22,26 +22,63 @@
 # the local_filestore implementation.
 
 
-def test_keyword_args():
-    """Tests that keyword args, and in particular 'parallel' are handled
-    correctly."""
-    filestore_path = 'gs://fake_dir'
-
-    with mock.patch('common.new_process.execute') as mocked_execute:
-        filestore_utils.rm(filestore_path, recursive=True, parallel=True)
-        mocked_execute.assert_called_with(
-            ['gsutil', '-m', 'rm', '-r', filestore_path], expect_zero=True)
-
-    with mock.patch('common.new_process.execute') as mocked_execute:
-        mocked_execute.return_value = new_process.ProcessResult(0, '', '')
-        filestore_utils.ls(filestore_path)
-        mocked_execute.assert_called_with(['gsutil', 'ls', filestore_path],
-                                          expect_zero=True)
-
-    filestore_path2 = filestore_path + '2'
-
-    with mock.patch('common.new_process.execute') as mocked_execute:
-        filestore_utils.cp(filestore_path, filestore_path2, parallel=True)
-        mocked_execute.assert_called_with(
-            ['gsutil', '-m', 'cp', filestore_path, filestore_path2],
-            expect_zero=True)
+class TestLocalFilestore:
+    """Tests for switching on local_filestore."""
+    FILESTORE = '/experiment_data'
+    DIR1 = '/dir1'
+    DIR2 = '/dir2'
+
+    def test_local_filestore_on(self, use_local_filestore):  # pylint: disable=unused-argument
+        """Tests that local_filestore switches on correctly."""
```

Comment:
Consider saying: 'Tests if local_filestore is used when experiment_filestore is a local path'

Reply:
@jonathanmetzman wdyt? same for the next one.

Reply:
The suggested comment is more consistent with current implementation, where `_using_gsutil()` function checks the format of experiment_filestore path.
However, this implementation is depended on two things:
(1) first, user set `local_experiment` as true
(2) during the read_and_valid_input function, it will check `experiment_filestore` format as Posix format if `local_experiment` is true.
(3) Then, when we use the filestore_utils, it checks the path.

In fact, it is not because of the path setting that introduces the  local running, but the `local_experiment` setting. Choosing the current implementation is a tradeoff and way to utilize current structure, where `experiment_filestore` is passed to all layers of running environments.

Reply:
We can switch filestore_utils to use the local impl if `LOCAL_EXPERIMENT` env var is set

Reply:
Right. Do we switch it in this PR?

Reply:
If we switch this on, maybe we can use `experiment_utils.is_local_experiment()` to replace the `_using_gsutil()`? And also pass environment variable `LOCAL_EXPERIMENT` to docker containers.

Reply:
This is also related to this issue: https://github.com/google/fuzzbench/issues/398

Reply:
Sorry, my comment was about making the docstring more clear: i.e., consider replacing " switches on correctly" with something more explicit, so it's more clear what it is testing exactly.